### PR TITLE
Refactor: Simplifying call model

### PR DESCRIPTION
**Pull request by ChriMarMe**
The purpose of this PR is to rework parts of the library to make it more maintainable and easier to extend with additional features.

### CODE REVIEW

**Comment by ChriMarMe** [comment]
I warp this PR up here, because anything related to a rework of sessions creates a behemoth of a PR and requires changes all over the repository. 

From this point on I'll only take reviews and suggestions for minor changes in the PR. Additional topics shall be brought forward via Issues.

File: cmd/sedlockctl/main.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"log"
 
 	"github.com/alecthomas/kong"
-	"github.com/open-source-firmware/go-tcg-storage/pkg/drive"
+	tcg "github.com/open-source-firmware/go-tcg-storage/pkg/core"
```
Comment (MDr164) [inline cmd/sedlockctl/main.go:11]: Is there a reason to import it as `tcg` here and in tcgdiskstat just as core?

File: pkg/core/table/admin.go [hunk 2]
```diff
@@ -11,25 +11,19 @@ import (
...
 )
 
 var (
-	Admin_TPerInfoTable = TableUID{0x00, 0x00, 0x02, 0x01, 0x00, 0x00, 0x00, 0x00}
-	Admin_C_PINTable    = TableUID{0x00, 0x00, 0x00, 0x0B, 0x00, 0x00, 0x00, 0x00}
-
 	Admin_C_PIN_ColumnPIN         uint = 3
 	Admin_SP_ColumnLifeCycleState uint = 6
 
 	// TODO: This is taken from the Opal spec, not sure how to dynamically find it...
```
Comment (MDr164) [inline pkg/core/table/admin.go:22]: You removed the code that this comment belonged to but not the comment itself, was that intended?

File: pkg/core/uid/methods.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package uid
```
Comment (MDr164) [inline pkg/core/uid/methods.go:1]: Missing License header above

File: pkg/core/uid/rows.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package uid
```
Comment (MDr164) [inline pkg/core/uid/rows.go:1]: Missing License header above

File: pkg/core/uid/tables.go [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package uid
```
Comment (MDr164) [inline pkg/core/uid/tables.go:1]: Missing License header above

File: pkg/core/uid/uid.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
+package uid
```
Comment (MDr164) [inline pkg/core/uid/uid.go:1]: Missing License header above

File: README.md [hunk 7]
```diff
@@ -73,33 +73,44 @@ is implementing.
...
-        }
-        // Session is now elevated
+	core, err := tcg.NewCore("/dev/sda")
+	if err != nil {
+		log.Fatalf("tcg.NewCore(/dev/sda) failed: %v",err)
+	}
+	defer core.Close()
+
+	// This will work if your drive implements GET_COMID,
+    // otherwise you will need to figure out the ComID and
```
Comment (MDr164) [inline README.md:92]: Weird indentation here

File: pkg/core/core.go [hunk 8]
```diff
@@ -92,3 +92,28 @@ func StackReset(d drive.DriveIntf, comID ComID) error {
 	}
 	return nil
 }
+
+func FindComID(d drive.DriveIntf, d0 *Level0Discovery) (ComID, ProtocolLevel, error) {
```
Comment (MDr164) [inline pkg/core/core.go:96]: Now that this function is exported, it should get a code comment stating its functionality.

**Review by MDr164 — CHANGES_REQUESTED** [review]
Looks mostly good to me. Just a few nitpicks here and there.

**Pull request merged**